templates/embed/embed.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Manage embed height **/
+
+// minimum height amp-iframes can be resized to
+var MIN_IFRAME_HEIGHT = 100;
+
+var preview = document.getElementById('preview');
+
+/**
+ * Set the preview width and height.
+ */
+function setPreviewDimension(width, height) {
+  preview.width = width;
+  preview.height = height;
+}
+
+/**
+ * Adjusts the iframe height based on the iframe's document height. The width is unchanged.
+ */
+function fitPreviewToContent() {
+  var iframeDoc = preview.contentDocument || preview.contentWindow.document;
+  setPreviewDimension(document.documentElement.clientWidth, iframeDoc.body.offsetHeight);
+}
+
+/**
+ * Post a message containing the document height to the amp-iframe to trigger the resize.
+ */
+function postEmbedHeightToViewer() {
+  var newHeight = document.getElementById('tabinterface').clientHeight;
+  if (newHeight < MIN_IFRAME_HEIGHT) {
+    console.log('embed height too small, reset height to 100px');
+    newHeight = MIN_IFRAME_HEIGHT;
+  }
+  // Tell the viewer about the new size
+  window.parent.postMessage({
```

Comment:
It is:

https://www.ampproject.org/docs/reference/components/amp-iframe#iframe-resizing
https://ampbyexample.com/components/amp-iframe/#resizable-iframes